src/single_instruction_test/heap.rs

```diff
@@ -47,6 +53,7 @@ impl<'a> Arbitrary<'a> for Heap {
 
 #[derive(Debug, Clone, Arbitrary)]
```

Comment:
You can no longer derive Arbitrary because `heap_id` should not be arbitrary. Make a constructor that takes heap_id instead.